docs/Migrating_LLB_to_a_PPL_based_Custom_Device.md

```diff
@@ -0,0 +1,203 @@
+
+The following image displays an example name and path of a Windows Configuration library file.
+
+![](images/Sys_Explorer_Path.PNG)
+
+ b) Custom Device Engine
+
+* The destination path should use the following structure: ``"..\built\Custom Device Name\Operating System``
+* The PLL file name should use the following format: [Custom Device Name] + [Engine] + [Operating System Name]
+
+The following image displays an example name and path of a Linux engine file.
+
+![](images/Engine_Path.PNG)
+
+**Note:** Starting with VeriStand 2021, VeriStand only supports a Linux x64 real-time operating system.
+
+##### Source Files Settings
+
+ a) Custom Device System Explorer
+
+In the Configuration Release PPL Properties dialog box, click **Source Files** and select the library containing the System Explorer files of your custom device (i.e. Configuration library). Set this library as the **Top-level Library**.
+
+![](images/PPL_Config_Source.png)
+
+**Note:** To distribute additional files with the PPL, such as a configuration XML file, add them to the **Always-Included** files list.
+
+ b) Custom Device Engine
+
+In the Configuration Release PPL Properties dialog box, click **Source Files** and select the library containing the engine files of your custom device (i.e. Configuration library). Set this library as the **Top-level Library**.
+
+![](images/Source_Files_Eng.png)
+
+**Note:** You should also configure PPLs for the Real Time targets build specifications.
+
+#### Update the XML
+
+You must update XML files either manually or by creating a **Post-Build Action** VI.
+
+##### Manual Updates
+
+To update manually, you will need to update the path for each LLB with the path of each corresponding PPL.
+
+The following image displays the XML code sequence for the custom device RT Driver VI on a Windows target.
+
+![](images/XML_Engine_Path_LLB.PNG)
+
+You need to update what comes after the `<Path>` tag.
+
+![](images/XML_Engine_Path_PPL.PNG)
+
+The `<Path>` tag contains a path for a VI located inside a PPL. You need to update all the `<Path>` tags in the XML that reference VIs inside the newly created PPLs.
+
+**Note:** You also need to update the `<RealTimeSystemDestination>` tags.
+
+##### Post-Build Action VI
+
+To automate path updates, implement a VI to update the XML file.
+
+The following image displays one possible VI.
+
+![](images/XML_updater.PNG)
+
+You can set this VI in LabVIEW as a Post-Build Action for your engine or configuration build specification.
+
+To set the action in LabVIEW, open the Engine Release PPL Properties dialog box and click **Pre/Post Build Actions** to enable **Excute VI after build**.
+
+![](images/Execute_XML_updater.PNG)
+
+For an example of such a VI being implemented, refer to the [FPGA Add-on Custom Device](https://github.com/ni/niveristand-fpga-addon-custom-device). this custom device will be available in the upcoming version of the NI VeriStand Custom Device Wizard.
```

Comment:
We should remove this section which references the FPGA Add on as an example, as the PR containing these changes was abandoned.